### ISSUE TITLE
Start installing codejail requirements from base.txt instead of py38.txt

### ISSUE THREAD

**Issue by timmc-edx** [edx/edx-arch-experiments#630]
The structure of the edxapp-sandbox-requirements has changed and we need to switch in order to keep getting updates: https://github.com/openedx/edx-platform/pull/34509

**First step**: Determine priority, may be able to lower it.

**Comment by kdmccormick** [comment]
Nice, you caught this before I got around to bugging you about it :) Just a heads up, [as I tried to explain here](https://github.com/openedx/edx-platform/tree/master/requirements/edx-sandbox#basetxt), we plan to keep requirements/edx/base.txt and requirements/edx-sandbox/base.txt synced in terms of constraints and language version. So, once Py3.8 support is dropped from edx-platform, it will be simultaneously be dropped for edx-sandbox. That might be exactly what you folks want, or it might not be, I'm not sure--just want to make sure you folks have all the information you need as you decide between `quince.txt` and `base.txt`.

**Comment by jristau1984** [comment]
Putting down a due date of "something like the Sumac release", so that we dont lose this.  The thought is that this can be added to a "pipeline" related epic in H2 2024.

**Comment by robrap** [comment]
Noting that it is entirely up to 2U when we want to get off an unsupported version of Python. It would be good for us to clean-up the unused file once we are done with it, but Axim does not plan on deleting this file. In other words, we don't have an external deadline for this work.

**Comment by robrap** [comment]
Additional questions:
1. Would it makes sense to move to Python 3.12 (rather than 3.11) to minimize the number of upgrades?
2. Would it make sense to bump Ubuntu versions to minimize upgrades? See https://github.com/openedx/edx-platform/issues/35314

**Comment by robrap** [comment]
Note: [Ubuntu 20.04 standard support runs out April 2025](https://ubuntu.com/about/release-cycle), and this upgrade work may need to be ticketed separately with SRE. I simply wanted to note that we need to take some action somewhere.

**Comment by robrap** [comment]
[inform] We missed the due date. Do we want to schedule this work?

**Comment by robrap** [comment]
FYI: @jristau1984

**Comment by jristau1984** [comment]
@robrap I believe that Arbi-BOM did upgrade the sandbox environment to Python 3.11 as a pre-requisite for several other testing efforts where sandboxes were needed.  This is the issue related to that effort: https://github.com/edx/edx-arch-experiments/issues/850.

Perhaps this is just a paperwork miss on my part?

**Comment by timmc-edx** [comment]
No, codejail is still running Python 3.8.10. (This is distinct from 2U's sandboxes; "sandbox" in the meaning of this ticket actually refers to codejail.)

**Comment by kdmccormick** [comment]
Separate from 2U's edx-sandbox 3.8->3.11 upgrade, the cleanup that we're hoping to do from the openedx side is just to delete [py38.txt](https://github.com/openedx/edx-platform/blob/master/requirements/edx-sandbox/py38.txt), which is just a simple pointer to [quince.txt](https://github.com/openedx/edx-platform/blob/master/requirements/edx-sandbox/releases/quince.txt). To avoid being broken by that cleanup, I recommend just updating your edx-sandbox deployment to point at `quince.txt` instead of `py38.txt`. You could then schedule the 3.8->3.11 edx-sandbox upgrade at your own pace with no openedx-based pressure.

**Comment by robrap** [comment]
PR: https://github.com/edx/configuration/pull/141

**Comment by robrap** [comment]
* When we land the PR I posted, we should be able to do the minor clean-up of removing the `py38.txt` file.
* Separately, Python 3.8 has gone out of support, so we'll still want to complete the upgrade.

**Comment by robrap** [comment]
@timmc-edx @jristau1984: We made the minor change in the issue description that freed up Kyle to delete the file. I am going to close this ticket, and we will take up our actual Python upgrade of codejail separately.
